frontend/simple/views/components/Chatroom/ChatMain.vue

```diff
@@ -0,0 +1,328 @@
+<template>
+  <div class="c-chatmain" :class="{ isActive: summary.title }">
+    <template v-if="summary.title">
+      <chat-header
+        :title="summary.title"
+        :description="summary.description"
+        routerBack="/messages"
+      >
+        <template slot="actions">
+          <button class="button is-icon">
+            <i class="fa fa-user-plus"></i>
+          </button>
+          <button class="button is-icon">
+            <i class="fa fa-bell"></i>
+          </button>
+          <menu-parent class="level-right">
+            <menu-trigger class="is-icon">
+              <i class="fa fa-ellipsis-v"></i>
+            </menu-trigger>
+
+            <!-- TODO later - be a drawer on mobile -->
+            <menu-content class="c-actions-content">
+              <list hasMargin>
+                <menu-item tag="button" itemId="hash-1" icon="heart">
+                  Option 1
+                </menu-item>
+                <menu-item tag="button" itemId="hash-2" icon="heart">
+                  Option 2
+                </menu-item>
+                <menu-item tag="button" itemId="hash-3" icon="heart">
+                  Option 3
+                </menu-item>
+              </list>
+            </menu-content>
+          </menu-parent>
+        </template>
+      </chat-header>
+
+      <div class="c-body is-flex" :style="bodyStyles">
+        <div v-if="details.isLoading">
+          LOADING... <!-- TODO Design a cool skeleton loading -->
+        </div>
+        <template v-else>
+          <div class="c-body-conversation" ref="conversation">
+            <conversation-greetings />
+            <component
+              v-for="(message, index) in details.conversation"
+              v-bind="getMessageAt[index]"
+            />
+            <message
+              v-for="(message, index) in ephemeral.pendingMessages"
+              v-bind="getPendingAt[index]"
```

Comment:
I do not understand this `getPendingAt[index]` and `getMessageAt[index]` (line 48).

Those are functions, not an array of data, so they should be indexable like that.

Reply:
No, those `computed` methods that return an array of data, each array element the data for each message...

Reply:
But that's not how you call a method. A method is called by doing `getPendingAt()[index]`

Reply:
But `getPendingAt ` is a `computed`, not a `method`.

Reply:
Ah... I see. Wow. I forgot about this JavaScript magic craziness that Vue.js does.